Core/NoOpRunner.Core/PlayerStates/PlayerState.cs

```diff
@@ -0,0 +1,24 @@
+﻿using NoOpRunner.Core.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NoOpRunner.Core.PlayerStates
+{
+    public abstract class PlayerState
+    {
+        protected PlayerOneStateMachine player;
+
+        public void SetPlayer(PlayerOneStateMachine player)
+        {
+            this.player = player;
+        }
```

Comment:
Why do you need this? Right now you only use it to write a log message.. If you want to differentiate running states by running direction, you should create multiple running state subclasses. State should be isolated, it shouldn't really care about it's owner.

Reply:
According to this:
https://refactoring.guru/design-patterns/state
The context class does know its owner, so I followed the same approach here. Logging running direction is only for demo during lab defense.

Reply:
Fair enough